packages/volto/src/components/manage/Widgets/SelectAutoComplete.jsx

```diff
@@ -50,215 +50,185 @@ const messages = defineMessages({
+        onChange={handleChange}
+        isMulti
+      />
+    </FormFieldWrapper>
+  );
+};
 
-  /**
-   * Render method.
-   * @method render
-   * @returns {string} Markup for the component.
-   */
-  render() {
-    const selectedOption = normalizeValue(
-      this.state.termsPairsCache,
-      this.props.value,
-      this.props.intl,
-    );
-    const SelectAsync = this.props.reactSelectAsync.default;
+SelectAutoComplete.propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  description: PropTypes.string,
+  required: PropTypes.bool,
+  error: PropTypes.arrayOf(PropTypes.string),
+  getVocabulary: PropTypes.func.isRequired,
+  choices: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+  ),
+  items: PropTypes.shape({
+    vocabulary: PropTypes.object,
+  }),
+  widgetOptions: PropTypes.shape({
+    vocabulary: PropTypes.object,
+  }),
+  value: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
+  ),
+  onChange: PropTypes.func.isRequired,
+  wrapped: PropTypes.bool,
+  isDisabled: PropTypes.bool,
+  placeholder: PropTypes.string,
+};
 
-    return (
-      <FormFieldWrapper {...this.props}>
-        <SelectAsync
-          id={`field-${this.props.id}`}
-          key={this.props.id}
-          isDisabled={this.props.disabled || this.props.isDisabled}
-          className="react-select-container"
-          classNamePrefix="react-select"
-          cacheOptions
-          defaultOptions={[]}
-          loadOptions={this.loadOptions}
-          onInputChange={(search) =>
-            this.setState({ searchLength: search.length })
-          }
-          noOptionsMessage={() =>
-            this.props.intl.formatMessage(
-              this.state.searchLength > this.SEARCH_HOLDOFF
-                ? messages.no_options
-                : messages.type_text,
-            )
-          }
-          styles={customSelectStyles}
-          theme={selectTheme}
-          components={{
-            ...(this.props.choices?.length > 25 && {
-              MenuList,
-            }),
-            MultiValueContainer,
-            ClearIndicator,
-            DropdownIndicator,
-            Option,
-          }}
-          value={selectedOption || []}
-          placeholder={
-            this.props.placeholder ??
-            this.props.intl.formatMessage(messages.select)
-          }
-          onChange={this.handleChange}
-          isMulti
-        />
-      </FormFieldWrapper>
-    );
-  }
-}
+SelectAutoComplete.defaultProps = {
+  description: null,
+  required: false,
+  items: {
+    vocabulary: null,
+  },
+  widgetOptions: {
+    vocabulary: null,
+  },
+  error: [],
+  choices: [],
+  value: null,
+};
 
-export const SelectAutoCompleteComponent = injectIntl(SelectAutoComplete);
+export const SelectAutoCompleteComponent = SelectAutoComplete;
 
 export default compose(
-  injectIntl,
   injectLazyLibs(['reactSelectAsync']),
   connect(
```

Comment:
The functional component should use getSelector and getDispatch instead of `connect`

Reply:
I'll close this PR to reduce noise in the list of PRs waiting for review. You can ask us to reopen it if you make this change.